packages/react-bindings/src/hooks/useStateManager.ts

```diff
@@ -58,19 +59,29 @@ const useStateManager = <
     ],
   })
 
+  // We need to keep the same reference to an object with actions to allow usage them as
+  // a dependency in useCallback() hook
+  Object.assign(latestActions, latestManager.current.actions)
+
+  // For development environments we disallow ability to extend object with other properties to
+  // avoid misusage
+  if (process.env.NODE_ENV !== 'production') {
+    if (Object.isExtensible(latestActions)) Object.preventExtensions(latestActions)
+  }
+
   // We need to pass exactly `manager.state` to provide the same state object during the same render
   // frame.
   // It keeps behavior consistency between React state tools and our managers
   // https://github.com/facebook/react/issues/11527#issuecomment-360199710
 
   if (process.env.NODE_ENV === 'production') {
-    return { state: latestManager.current.state, actions: latestManager.current.actions }
+    return { state: latestManager.current.state, actions: latestActions }
```

Comment:
As the two return statements now differ only in state I would rather change them to single return with ternary inside